pkg/agent/route/route.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package route
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/config"
+	"github.com/vmware-tanzu/antrea/pkg/agent/iptables"
+	"github.com/vmware-tanzu/antrea/pkg/agent/types"
+	"github.com/vmware-tanzu/antrea/pkg/agent/util"
+)
+
+const (
+	// AntreaServiceTable is route table name for Antrea service traffic.
+	AntreaServiceTable = "Antrea-service"
+	// AntreaServiceTableIdx is route table index for Antrea service traffic.
+	AntreaServiceTableIdx = 300
+)
+
+// Client is route client.
+type Client struct {
+	nodeConfig *types.NodeConfig
+}
+
+type serviceRtTableConfig struct {
+	Idx  int
+	Name string
+}
+
+func (s *serviceRtTableConfig) String() string {
+	return fmt.Sprintf("%s: idx %d", s.Name, s.Idx)
+}
+
+func (s *serviceRtTableConfig) IsMainTable() bool {
+	return s.Name == "main"
+}
+
+var (
+	// ServiceRtTable contains Antrea service route table information.
+	ServiceRtTable = &serviceRtTableConfig{Idx: 254, Name: "main"}
+)
+
+// NewClient returns a route client
+func NewClient() *Client {
+	return &Client{}
+}
+
+// Initialize sets up route tables for Antrea.
+func (c *Client) Initialize(nodeConfig *types.NodeConfig) error {
+	c.nodeConfig = nodeConfig
+
+	if config.TrafficEncapMode.SupportsNoEncap() {
+		ServiceRtTable.Idx = AntreaServiceTableIdx
+		ServiceRtTable.Name = AntreaServiceTable
+	}
+
+	if ServiceRtTable.IsMainTable() {
+		_ = c.RemoveServiceRouting()
+		return nil
+	}
+	f, err := os.OpenFile("/etc/iproute2/rt_tables", os.O_RDWR|os.O_APPEND, 0)
```

Comment:
What about using a constant for file path `/etc/iproute2/rt_tables`?
Another question, does this file need to be mounted from the host?

Reply:
this file requires no mounting, it exits in each network namespace.

Reply:
used a constant for routeConfigPath. done